src/conversions/chrono.rs

```diff
@@ -729,6 +730,30 @@ mod tests {
         })
     }
 
+    #[test]
+    fn test_pyo3_offset_fixed_frompyobject_created_in_python() {
+        Python::with_gil(|py| {
+            let code = r#"
+import datetime
+t = datetime.datetime.fromtimestamp(12345).replace(tzinfo=datetime.timezone(datetime.timedelta(hours=-8)))
+            "#.trim();
+
+            let globals = PyDict::new(py);
+            py.run(code, Some(globals), None).unwrap();
```

Comment:
I think this might look simpler by using [`eval`](https://docs.rs/pyo3/latest/pyo3/marker/struct.Python.html#method.eval)?

Reply:
If you structure the code with `eval`, how do you get the import? Maybe I am misunderstanding

Reply:
A useful trick is to use `__import__` as for example [here](https://github.com/PyO3/rust-numpy/blob/08510a3aa31a09c45df4602c484ac81056f0c4e8/src/array.rs#L742).

Reply:
I think you'll wind up with an expression that looks like

```
let code = format!("__import('datetime').datetime.fromtimestamp({timestamp}).replace(tzinfo=__import('datetime').timezone(__import('datetime').timedelta(hours=-8)))");
```

Note the 3 uses of `__import__` to make this work, IMO the `run` solution is more readable

Reply:
How about

```rust
let globals = [("datetime", py.import("datetime").unwrap())].into_py_dict(py);
let t = py.eval(format!("datetime.datetime.fromtimestamp({timestamp}).replace(tzinfo=datetime.timezone(datetime.timedelta(hours={hours})))"), Some(globals), None).unwrap();
```

then?

Reply:
(Also note that you cannot use inline format literals due to our MSRV.)